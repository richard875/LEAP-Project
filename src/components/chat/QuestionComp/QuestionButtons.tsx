import { faTrashCan } from "@fortawesome/free-regular-svg-icons";
import { faCheck, faPenToSquare } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import {
  AlertDialog,
  AlertDialogAction,
  AlertDialogCancel,
  AlertDialogContent,
  AlertDialogDescription,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogTitle,
  AlertDialogTrigger,
} from "@/components/ui/alert-dialog";
import {
  Tooltip,
  TooltipContent,
  TooltipProvider,
  TooltipTrigger,
} from "@/components/ui/tooltip";

const QuestionButtons = ({
  editing,
  handleEdit,
  handleDelete,
}: {
  editing: boolean;
  handleEdit: () => Promise<void>;
  handleDelete: () => Promise<void>;
}) => {
  return (
    <div className="mt-1 flex gap-1">
      <TooltipProvider>
        <Tooltip>
          <TooltipTrigger>
            <div
              onClick={handleEdit}
              className="w-7.5 h-7.5 rounded-lg flex items-center justify-center hover:bg-neutral-100 dark:hover:bg-neutral-800 transition cursor-pointer"
            >
              <FontAwesomeIcon
                icon={editing ? faCheck : faPenToSquare}
                className="brightness-95 text-sm"
              />
            </div>
          </TooltipTrigger>
          <TooltipContent className="bg-foreground text-background">
            <p>{editing ? "Done" : "Edit"}</p>
          </TooltipContent>
        </Tooltip>
      </TooltipProvider>

      <TooltipProvider>
        <Tooltip>
          <AlertDialog>
            <TooltipTrigger asChild>
              <AlertDialogTrigger asChild>
                <button
                  type="button"
                  className="w-7.5 h-7.5 rounded-lg flex items-center justify-center hover:bg-neutral-100 dark:hover:bg-neutral-800 transition cursor-pointer"
                >
                  <FontAwesomeIcon
                    icon={faTrashCan}
                    className="brightness-95 text-sm"
                  />
                </button>
              </AlertDialogTrigger>
            </TooltipTrigger>

            <TooltipContent className="bg-foreground text-background">
              <p>Delete</p>
            </TooltipContent>

            <AlertDialogContent className="rounded-2xl">
              <AlertDialogHeader className="select-none">
                <AlertDialogTitle>
                  Are you sure you want to delete this question?
                </AlertDialogTitle>
                <AlertDialogDescription>
                  This action cannot be undone. This will permanently delete
                  this question and all of its responses.
                </AlertDialogDescription>
              </AlertDialogHeader>
              <AlertDialogFooter>
                <AlertDialogCancel className="cursor-pointer select-none rounded-lg">
                  Cancel
                </AlertDialogCancel>
                <AlertDialogAction
                  onClick={handleDelete}
                  className="bg-red-500 hover:bg-red-600 text-white cursor-pointer select-none rounded-lg"
                >
                  Delete
                </AlertDialogAction>
              </AlertDialogFooter>
            </AlertDialogContent>
          </AlertDialog>
        </Tooltip>
      </TooltipProvider>
    </div>
  );
};

export default QuestionButtons;
