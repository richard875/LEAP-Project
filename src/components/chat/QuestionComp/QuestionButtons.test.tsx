import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";

import QuestionButtons from "./QuestionButtons";

describe("QuestionButtons", () => {
  it("should render correctly", () => {
    const { container } = render(
      <QuestionButtons
        editing={false}
        handleEdit={async () => {}}
        handleDelete={async () => {}}
      />
    );
    expect(container).toBeTruthy();
  });

  it("should match the snapshot", () => {
    const { asFragment } = render(
      <QuestionButtons
        editing={false}
        handleEdit={async () => {}}
        handleDelete={async () => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the correct elements in the component", () => {
    render(
      <QuestionButtons
        editing={false}
        handleEdit={async () => {}}
        handleDelete={async () => {}}
      />
    );

    const buttons = screen.getAllByRole("button");
    expect(buttons.length).toBe(2);
  });
});
