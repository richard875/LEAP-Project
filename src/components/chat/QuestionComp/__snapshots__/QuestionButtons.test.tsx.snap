// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionButtons should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mt-1 flex gap-1"
  >
    <button
      data-slot="tooltip-trigger"
      data-state="closed"
    >
      <div
        class="w-7.5 h-7.5 rounded-lg flex items-center justify-center hover:bg-neutral-100 dark:hover:bg-neutral-800 transition cursor-pointer"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-pen-to-square brightness-95 text-sm"
          data-icon="pen-to-square"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M471.6 21.7c-21.9-21.9-57.3-21.9-79.2 0L362.3 51.7l97.9 97.9 30.1-30.1c21.9-21.9 21.9-57.3 0-79.2L471.6 21.7zm-299.2 220c-6.1 6.1-10.8 13.6-13.5 21.9l-29.6 88.8c-2.9 8.6-.6 18.1 5.8 24.6s15.9 8.7 24.6 5.8l88.8-29.6c8.2-2.7 15.7-7.4 21.9-13.5L437.7 172.3 339.7 74.3 172.4 241.7zM96 64C43 64 0 107 0 160L0 416c0 53 43 96 96 96l256 0c53 0 96-43 96-96l0-96c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7-14.3 32-32 32L96 448c-17.7 0-32-14.3-32-32l0-256c0-17.7 14.3-32 32-32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L96 64z"
            fill="currentColor"
          />
        </svg>
      </div>
    </button>
    <button
      aria-controls="radix-«r7»"
      aria-expanded="false"
      aria-haspopup="dialog"
      class="w-7.5 h-7.5 rounded-lg flex items-center justify-center hover:bg-neutral-100 dark:hover:bg-neutral-800 transition cursor-pointer"
      data-slot="tooltip-trigger"
      data-state="closed"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-trash-can brightness-95 text-sm"
        data-icon="trash-can"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M170.5 51.6L151.5 80l145 0-19-28.4c-1.5-2.2-4-3.6-6.7-3.6l-93.7 0c-2.7 0-5.2 1.3-6.7 3.6zm147-26.6L354.2 80 368 80l48 0 8 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-8 0 0 304c0 44.2-35.8 80-80 80l-224 0c-44.2 0-80-35.8-80-80l0-304-8 0c-13.3 0-24-10.7-24-24S10.7 80 24 80l8 0 48 0 13.8 0 36.7-55.1C140.9 9.4 158.4 0 177.1 0l93.7 0c18.7 0 36.2 9.4 46.6 24.9zM80 128l0 304c0 17.7 14.3 32 32 32l224 0c17.7 0 32-14.3 32-32l0-304L80 128zm80 64l0 208c0 8.8-7.2 16-16 16s-16-7.2-16-16l0-208c0-8.8 7.2-16 16-16s16 7.2 16 16zm80 0l0 208c0 8.8-7.2 16-16 16s-16-7.2-16-16l0-208c0-8.8 7.2-16 16-16s16 7.2 16 16zm80 0l0 208c0 8.8-7.2 16-16 16s-16-7.2-16-16l0-208c0-8.8 7.2-16 16-16s16 7.2 16 16z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
